docker/docker.go

```diff
@@ -0,0 +1,51 @@
+package docker
+
+import (
+	"io"
+	"log"
+	"os/exec"
+)
+
+const volumePattern string = "coduno-volume"
+
+// DockerCmd holds the cmd to be executed as well as the executed
+// image and the volume from wich it gets data
+type DockerCmd struct {
+	exec.Cmd
+	image,
+	volume string
+}
+
+// NewDockerCmd creates a command for a docker run
+func NewDockerCmd(language, volume string) DockerCmd {
+	image := "coduno/fingerprint-" + language
+	return DockerCmd{
+		*exec.Command(
+			"docker",
+			"run",
+			"--rm",
+			"-v",
+			volume+":/run",
+			image),
+		image,
+		volume,
+	}
+}
+
+// GetStreams returns the in, out and err streams from a command
+func GetStreams(cmd *exec.Cmd) (outCmd, errCmd io.ReadCloser, inCmd io.WriteCloser, err error) {
+	outCmd, err = cmd.StdoutPipe()
+	if err != nil {
+		log.Fatal(err)
```

Comment:
In the older version, calling `log.Fatal` made sense, because it was in `main` and the program was not run as a server, just invoked from a Git hook. Therefore it would just exit. By moving to a server scenario with cfd4fc040dfe03fc4087e31d0fb99eb55fd1e674 we should have fixed that.